### ISSUE TITLE
Changing entry.py for MisconfigurationException error

### ISSUE THREAD

**Issue by Sangyoon-Bae** [microsoft/Graphormer#40]
Hi! This is Stella from Seoul National University, I'm getting a lot of help from your code.
I have a question about entry.py line 87.
Originally it has **metric = 'valid_' + get_dataset(dm.dataset_name)['metric']**
<img width="403" alt="image" src="https://user-images.githubusercontent.com/90450600/144160119-36edf968-034e-4520-89ed-6eac2cbb4a50.png">
but when I run model, I faced error like this:
'pytorch_lightning.utilities.exceptions.MisconfigurationException: ModelCheckpoint(monitor='valid_mae') not found in the returned metrics: ['train_loss']. HINT: Did you call self.log('valid_mae', value) in the LightningModule?'

So I changed the line 87 as **metric = 'train_loss'**
<img width="434" alt="image" src="https://user-images.githubusercontent.com/90450600/144160272-e77ced0c-cd1d-4910-899f-9760cbbab5e9.png">
and it runs well.

I'm quite afraid that I'm doing something wrong, is it right way to modify the code?
Here are some useful information for my project:
1. task: regression
2. input type : integer (originally continuous value, but discretized)
3. target type : real value
4. eval metric : rmse
5. features from data.py
-             'num_class': 1,
-             'loss_fn': F.l1_loss,
-             'metric': 'mae',
-             'metric_mode': 'min',

**Comment by zhengsx** [comment]
You can have a look about the usage of ModelCheckpoint in pytorch_lightning (see [here](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.model_checkpoint.html#model-checkpointing)). Basically it helps to monitor a self-defined [metric](https://github.com/microsoft/Graphormer/blob/main/graphormer/model.py#L257) during the training process.

**Comment by Sangyoon-Bae** [comment]
Thank you for kind reply! I'm always grateful for your help :)
I used shell script in [training ZINC dataset](https://github.com/microsoft/Graphormer/tree/main/examples/benchmarking-gnns), and is this script just for pre-training?
If I want to validate model and test it, then should I do transfer learning like [molhiv training example](https://github.com/microsoft/Graphormer/blob/main/examples/ogb/hiv.sh) with pre-trained model which is made with [this script](https://github.com/microsoft/Graphormer/tree/main/examples/benchmarking-gnns)?

**Comment by zhengsx** [comment]
The reproduction of the ZINC results reported in our paper doesn't need pre-training. You can directly use the script. Btw, remember to test your model on ZINC testset, refer to #35 .

**Comment by Sangyoon-Bae** [comment]
Then [shell script in training ZINC dataset ](https://github.com/microsoft/Graphormer/tree/main/examples/benchmarking-gnns) do training and validation, and should I use test dataset with another script to reproduce test MAE 0.122?

<img width="719" alt="why ends so early" src="https://user-images.githubusercontent.com/90450600/145065365-88db2664-7d8a-4518-9009-df7f00340fe2.png">
then loss in log bar (it is not ZINC data) means validation loss?

**Comment by Sangyoon-Bae** [comment]
I solved the second problem! the loss in log bar means average loss about the past 100 batches in training.

**Comment by zhengsx** [comment]
> Then [shell script in training ZINC dataset ](https://github.com/microsoft/Graphormer/tree/main/examples/benchmarking-gnns) do training and validation, and should I use test dataset with another script to reproduce test MAE 0.122?
> 
> <img alt="why ends so early" width="719" src="https://user-images.githubusercontent.com/90450600/145065365-88db2664-7d8a-4518-9009-df7f00340fe2.png">
> 
> then loss in log bar (it is not ZINC data) means validation loss?

Yes, pls save your ckpt according to best val then test it on Testset.

**Comment by zhengsx** [comment]
Close this issue due to inactive. Feel free to raise a new one or reopen this one for any further question.
